lib/ProxySQL_Cluster.cpp

```diff
@@ -100,7 +100,7 @@ void * ProxySQL_Cluster_Monitor_thread(void *args) {
 			//mysql_options(conn, MYSQL_OPT_WRITE_TIMEOUT, &timeout);
 			{ unsigned char val = 1; mysql_options(conn, MYSQL_OPT_SSL_ENFORCE, &val); }
 			//rc_conn = mysql_real_connect(conn, node->hostname, username, password, NULL, node->port, NULL, CLIENT_COMPRESS); // FIXME: add optional support for compression
-			rc_conn = mysql_real_connect(conn, node->hostname, username, password, NULL, node->port, NULL, 0);
+			rc_conn = mysql_real_connect(conn, MySQL_Monitor::dns_lookup(node->hostname).c_str(), username, password, NULL, node->port, NULL, 0);
```

Comment:
The approach here is correct, but I think we now have the opportunity to improve cluster even further.
If more then *one* IP is present for a given ProxySQL node, we should generate a warning in the error log. Perhaps even an error!
Ideally we should also prevent to establish a connection to that IP , but this is perhaps too much.